tasks/create-advisory/create-advisory.yaml

```diff
@@ -72,18 +70,73 @@ spec:
 
         pipelinerun_label="internal-services.appstudio.openshift.io/pipelinerun-uid"
 
+        # only 2 gitlab instances are permitted...prod and staging
+        # these are their secret names
+        #
+        prodSecretName="create-advisory-secret"
+        stagingSecretName="create-advisory-stage-secret"
+        #
+        # detect which one to use based on repositories specified
+        #
+        pending_repositories=$(jq -r '.spec.data.mapping.components[] | select(.repository | contains("quay.io/redhat-pending/")) | .repository' $(workspaces.data.path)/$(params.releasePlanAdmissionPath))
+        prod_repositories=$(jq -r '.spec.data.mapping.components[] | select(.repository | contains("quay.io/redhat-prod/")) | .repository' $(workspaces.data.path)/$(params.releasePlanAdmissionPath))
+        orphan_repositories=$(jq -r '.spec.data.mapping.components[] | select(.repository | contains("quay.io/redhat-prod/") | not) | select(.repository | contains("quay.io/redhat-pending/") | not) | .repository' $(workspaces.data.path)/$(params.releasePlanAdmissionPath))
+
+        foundPendingRepositories=false
+        if [ ! -z "${pending_repositories[@]}" ]; then
+          foundPendingRepositories=true
+        fi
+
+        foundProdRepositories=false
+        if [ ! -z "${prod_repositories[@]}" ]; then
+          foundProdRepositories=true
+        fi
+
+        foundOrphanRepositories=false
+        if [ ! -z "${orphan_repositories[@]}" ]; then
+          foundOrphanRepositories=true
+        fi
+
+        echo "foundPendingRepositories: ${foundPendingRepositories}"
+        echo "foundProdRepositories ${foundProdRepositories}"
+        echo "foundOrphanRepositories ${foundOrphanRepositories}"
+
+        if [ "${foundPendingRepositories}" == "true" ] && [ "${foundProdRepositories}" == "true" ]; then
+          echo "Error: cannot publish to both redhat-pending and redhat-prod repositories"
+          exit 1
+        fi
+
+        if [ "${foundPendingRepositories}" == "false" ] && [ "${foundProdRepositories}" == "false" ]; then
+          echo "Error: you must publish to either redhat-pending or redhat-prod repositories"
+          exit 1
+        fi
+
+        if [ "${foundOrphanRepositories}" == "true" ]; then
+          echo "Error: you must publish to either redhat-pending or redhat-prod repositories"
+          exit 1
+        fi
+
+        # at this point, one of foundPendingRepositories or foundProdRepositories
+        # is true.
+        #
+        advisorySecretName="${prodSecretName}"
+        if [ "${foundPendingRepositories}" == "true" ]; then
+          advisorySecretName="${stagingSecretName}"
+        fi
+
         echo "Creating InternalRequest to create advisory..."
         internal-request -r "$(params.request)" \
                          -p application="${application}" \
                          -p origin="${origin}" \
                          -p repo="${repo}" \
```

Comment:
this can be removed now (as well as l65) IIUC

Reply:
I thought I would keep it for now...until changes are deployed via the internal services catalog. Then I'll clean it up

Reply:
Okay, that probably makes things easier